crates/primitives/simulations/src/lib.rs

```diff
@@ -5,19 +5,33 @@ pub extern crate alloc;
 
 use alloc::vec::Vec;
 
+use blockifier::transaction::errors::TransactionExecutionError;
 use blockifier::transaction::objects::TransactionExecutionInfo;
+use starknet_api::api_core::ContractAddress;
 use starknet_core::types::SimulationFlag;
 
 // TODO: This is a placeholder
 // https://github.com/starkware-libs/starknet-specs/blob/master/api/starknet_api_openrpc.json#L3919
 // The official rpc expect use to return the trace up to the point of failure.
 // Figuring out how to get that is a problem for later
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 #[cfg_attr(feature = "scale-info", derive(scale_info::TypeInfo))]
 #[cfg_attr(feature = "parity-scale-codec", derive(parity_scale_codec::Encode, parity_scale_codec::Decode))]
-pub struct PlaceHolderErrorTypeForFailedStarknetExecution;
+pub enum Error {
+    ContractNotFound(ContractAddress),
+    // TODO: replace string with Blockifier TransactionExecutionError
```

Comment:
When will thins be possible?

Reply:
When Blockifier and Cairo-vm add subtrate encode, decode and typeinfo to the error type. Right now I just wrap wrap the string formatting and pipe through to the RPC

Reply:
created #1579